x-pack/plugin/core/src/main/resources/monitoring-apm-server.json

```diff
@@ -0,0 +1,686 @@
+{
+  "index_patterns": [
+    ".monitoring-apm-server-${monitoring.template.version}-*"
+  ],
+  "settings": {
+    "index.auto_expand_replicas": "0-1",
+    "index.codec": "best_compression",
+    "index.format": 6,
+    "index.number_of_replicas": 0,
+    "index.number_of_shards": 1
+  },
+  "version": 7000001,
+  "mappings": {
+    "doc": {
+      "dynamic": false,
+      "properties": {
+        "beats_state": {
+          "properties": {
+            "beat": {
+              "properties": {
+                "host": {
+                  "type": "keyword"
+                },
+                "name": {
+                  "type": "keyword"
```

Comment:
thoughts on marking this and other fields not relevant to apm-server as `enabled: false` ?

Reply:
Or we could even just remove such fields from the mapping entirely for now, and add them as/when we need them. WDYT?

Reply:
Would that require removing them from the libbeat reporting for apm-server?

Reply:
No, I don't think so because [the mapping uses `dynamic: false`](https://github.com/elastic/elasticsearch/pull/32515/files#diff-36fe8240d5b097ab48d090f06a74e812R15), so any un-mapped fields will be simply ignored and not indexed (as opposed to if the mapping were to use `dynamic: strict`, in which case un-mapped fields would cause errors).

Reply:
The belief is right. Anything unmapped just tags along in the `_source` but is not used by ES.

Reply:
So @graphaelli, which fields specifically is it okay to remove from the mapping? Let me know and I'll take them out of this PR!

Reply:
all of `metrics.libbeat.config.*` can go as we don't support modules or reloading

Reply:
@ycombinator Perhaps we can also make it possible to not show up in the events if not there.